### ISSUE TITLE
[Contao 4] post sale always logs exception (even on success)

### ISSUE THREAD

**Issue by fritzmg** [isotope/core#1834]
If the post sale request is successful, isotope simply creates an empty `Haste\Http\Response\Response` [here](https://github.com/isotope/core/blob/2.4.3/system/modules/isotope/postsale.php#L165-L166).

However, this is a problem in Contao 4, since this is done within a try-catch block, that catches __all__ exceptions. In Contao 4 `Haste\Http\Response\Response::send()` creates a `Contao\CoreBundle\Exception\ResponseException`. See [here](https://github.com/codefog/contao-haste/blob/4.17.3/library/Haste/Http/Response/Response.php#L258-L262):
```php
if ($blnExit && class_exists('Contao\CoreBundle\Exception\ResponseException')) {
    throw new ResponseException(
        new SymfonyResponse($this->strContent, $this->intStatus, $this->arrHeaders)
    );
}
```
Thus any post sale request will create a
```
Exception in post-sale request. See system/logs/isotope_postsale.log for details.
```
log entry, even when everything was successful.

**Comment by Toflar** [comment]
I think this should be enough:

```diff
} catch (\Exception $e) {
+
+ if (is_a($e, 'Contao\CoreBundle\Exception\ResponseException', true)) {
+     throw $e;
+ }

\System::log(
```

Imho `is_a()` does not trigger an autoload in that case and it should thus work in Contao 3.5 as well. But needs to be tested.

**Comment by fritzmg** [comment]
Shouldn't it be
```php
is_a($e, 'Contao\CoreBundle\Exception\ResponseException', true)
```
or rather
```php
is_a($e, 'Contao\CoreBundle\Exception\ResponseException')
```
?

The [documention](http://php.net/manual/en/function.is-a.php) states

> If this parameter set to __FALSE__, string class name as _object_ is not allowed. This also prevents from calling autoloader if the class doesn't exist.

**Comment by aschempp** [comment]
Fixed in f0afb9d18642bebdad84eafe2647d8f6a3c89363

The second parameter does not apply to us at all, because it's related to the first parameter which is always an object in our case.
